### DISCUSSION TITLE
Sample MauiAppClient - Empty?

### DISCUSSION THREAD

**Discussion by Sean-URS** [Mimetis/Dotmim.Sync#892] [Q&A]
Hi, has anyone in the Dotmim community shared updates to the MauiAppClient sample?
The MauiAppClient in the samples folder appears to be a basic Maui template with no code related to working with Dotmim.Sync. 

I am hoping to use the framework to sync data from a server (SQL) to multiple client applications (SQLite) that are built with Maui Blazor.  

I have been using the “Progress” (is indicated in issue #859) and not the “SynchronousProgress”, however have not managed to either find the local SQLite database on the Android emulator or get any feedback from the Progress, no errors are thrown.

Thanks for a well-documented and very useful framework.

**Comment by Sean-URS** [comment]
Have made some progress on this... the "Progress" now returns the error "cleartext http traffic to 10.0.2.2 not permitted"
when using an http url for the server.
It is easier working with Android Emulators in a dev environment with http so I had set up the server to have both http and https   
When using the https apiUrl the following error is returned
"java.security.cert.CertPathValidatorException: Trust anchor for certification path not found."

The next step I'm going to work on is setting the Android Emulator to use secure connections (https)
Hope I'm heading in the correct direction.

**Reply by Mimetis** [reply]
Hey Sean, I've made a sample, you can access here https://github.com/Mimetis/Dotmim.Sync/tree/master/Samples/MAUI

I've just tested the Android version with an emulator running locally.
Let me know if it's enough

**Reply by Mimetis** [reply]
For unsecure http; you need to use a [network xml config](
https://github.com/Mimetis/Dotmim.Sync/blob/master/Samples/MAUI/MauiAppClient/Platforms/Android/Resources/xml/network_security_config.xml) where you are specifying `cleartextTrafficPermitted="true"`

This xml file is referenced in [Android Manifest](https://github.com/Mimetis/Dotmim.Sync/blob/master/Samples/MAUI/MauiAppClient/Platforms/Android/AndroidManifest.xml)

Don't forget to use the `http://10.0.2.2` special url to access the `localhost` endpoint

Don't forget to specify a server check certificate as well

``` cs
#if DEBUG
            if (DeviceInfo.Platform == DevicePlatform.Android && syncApiUri.Host == "10.0.2.2")
            {
                handler.ServerCertificateCustomValidationCallback = (message, cert, chain, errors) =>
                {
                    if (cert.Issuer.Equals("CN=localhost"))
                        return true;
                    return errors == System.Net.Security.SslPolicyErrors.None;
                };
            }

            if (DeviceInfo.Platform == DevicePlatform.Android && syncApiUri.Host == "10.0.2.2")
                this.httpClient.DefaultRequestHeaders.Host = $"localhost:{syncApiUri.Port}";
#endif

```
A lot of little things :)

**Reply by Sean-URS** [reply]
Many thanks, this is most helpful. Managed to get my test database of 100 000 products to sync.
![image](https://user-images.githubusercontent.com/113436819/206457393-533f140d-2f9d-478c-8687-48bcec26e252.png)

**Reply by Mimetis** [reply]
Nice !!!
